core/templates/domain/topics_and_skills_dashboard/topics-and-skills-dashboard-backend-api.service.ts

```diff
@@ -17,27 +17,34 @@
   from the backend and to merge skills from the dashboard.
  */
 
-require(
-  'domain/topics_and_skills_dashboard/' +
-  'topics-and-skills-dashboard-domain.constants.ajs.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { TopicsAndSkillsDashboardDomainConstants } from
+  // eslint-disable-next-line max-len
+  '../topics_and_skills_dashboard/topics-and-skills-dashboard-domain.constants';
 
-angular.module('oppia').factory('TopicsAndSkillsDashboardBackendApiService', [
-  '$http', 'MERGE_SKILLS_URL', function($http, MERGE_SKILLS_URL) {
-    var _fetchDashboardData = function() {
-      return $http.get('/topics_and_skills_dashboard/data');
-    };
 
-    var _mergeSkills = function(oldSkillId, newSkillId) {
-      var mergeSkillsData = {
-        old_skill_id: oldSkillId,
-        new_skill_id: newSkillId
-      };
-      return $http.post(MERGE_SKILLS_URL, mergeSkillsData);
-    };
+@Injectable({
+  providedIn: 'root'
+})
+export class TopicsAndSkillsDashboardBackendApiService {
+  constructor(private http: HttpClient) {}
+
+  fetchDashboardData(): Promise<any> {
```

Comment:
Is it possible to use a more specific type here? We want to avoid using `any`.

Reply:
I think we can use Object type here .

Reply:
Let's just introduce an interface instead.

```typescript
interface ITopicSummaryBackendDict {
  /* eslint-disable camelcase */
  id: string;
  additional_story_count: number;
  uncategorized_skill_count: number;
  topic_model_created_on: number;
  // And so on...
  /* eslint-enable camelcase */
}

interface ISkillSummaryBackendDict {
  /* eslint-disable camelcase */
  // As before...
  /* eslint-enable camelcase */
}

interface ITopicsAndSkillsDashboardDataBackendDict {
  topic_summary_dicts: ITopicSummaryBackendDict[];
  skill_summary_dicts: ISkillSummaryBackendDict[];
  // And so on...
}
```

Reply:
Yep, let's do that! Great suggestion @brianrodri!